### PR TITLE
Remove pip-tools from base requirements and upgrade requirements

### PR DESCRIPTION
**Pull request by morenol**
See https://openedx.atlassian.net/projects/BOM/issues/BOM-1359.

This error message is thrown while upgrading edx-analytics-data-api: **edx-enterprise-data 1.3.13 has requirement pip-tools==4.3.0, but you'll have pip-tools 4.5.1 which is incompatible.** See: https://github.com/edx/edx-analytics-data-api/pull/299

Since pip-tools is not an actual requierement of edx-enterprise-data I am removing that from base.in and upgrading the requirements. Also it was removed tox from base.in requirements.

Also it was added a contraints.txt file to accomplish the OEP-18 practices and a pip-tools.in file to keep the version of the tools used to upgrade the dependencies.

### Constraints changed

Added 
edx-django-utils<3.0.0
edx-rest-api-client<4.0.0

tox-battery from 0.5.1 to 0.5.2

[tox](https://github.com/tox-dev/tox/blob/master/docs/changelog.rst) from 3.10.0 to none

### Reviewers
- [x] @andrey-canon
- [x] Is this ready for edX's review?
- [ ] @jmbowman 
### Notes
Some constraints were added due to the lack of support of python 2.7, Is it ok to add them? (edx-django-utils, edx-rest-api-client)

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @morenol! I've created [OSPR-4135](https://openedx.atlassian.net/browse/OSPR-4135) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: requirements/constraints.txt [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# that will be applied if a package is needed.
+#
+# When pinning something here, please provide an explanation of why.  Ideally,
+# link to other information that will help people in the future to remove the
+# pin when possible.  Writing an issue against the offending project and
+# linking to it here is good.
+
+# TODO: Many pinned dependencies should be unpinned and/or moved to this constraints file.
+
+# Pinned for PGPy, see https://github.com/SecurityInnovation/PGPy/issues/21
```
Comment (jmbowman) [inline requirements/constraints.txt]: Did you mean https://github.com/SecurityInnovation/PGPy/issues/217 ?  That seems to have been fixed by pgpy 0.5.0, and we're already using 0.5.2 here.

File: requirements/test-reporting.in [hunk 2]
```diff
@@ -1,4 +1,6 @@
-tox==3.0.0
+-c constraints.txt
+
+tox
 tox-battery==0.5.1
```
Comment (jmbowman) [inline requirements/test-reporting.in]: We should probably remove this version constraint, 0.5.2 fixed compatibility with the latest versions of pip: https://github.com/signalpillar/tox-battery/pull/20

**Comment by jmbowman** [comment]
One more thing: setup.py should be loading requirements from `base.in`, not `base.txt`.  We don't want to be too strict with exact dependency versions for library packages.

File: requirements/test-reporting.in (hunk 2 above)
Comment (morenol) [reply requirements/test-reporting.in]: Done

File: requirements/base.txt [hunk 3]
```diff
@@ -6,83 +6,71 @@
...
-cryptography==2.8
+cryptography==2.8         # via -r requirements/base.in, django-fernet-fields, paramiko, pgpy, pyopenssl
 django-crum==0.7.5        # via edx-rbac
-django-fernet-fields==0.5
-django-model-utils==3.2.0
-django-waffle==0.18.0     # via edx-django-utils, edx-drf-extensions
-django==1.11.27
+django-fernet-fields==0.5  # via -r requirements/base.in
+django-model-utils==3.2.0  # via -r requirements/base.in, edx-rbac
+django-waffle==0.20.0     # via edx-django-utils, edx-drf-extensions
```
Comment (jmbowman) [inline requirements/base.txt]: Version 0.19.0 dropped support for Django 2.0 and 2.1, which we want to test with before upgrading to Django 2.2.  We should probably put an entry in `constraints.txt` for this.

File: requirements/test-master.txt [hunk 4]
```diff
@@ -8,101 +8,98 @@ amqp==1.4.9               # via kombu
...
+ddt==1.2.2                # via -r requirements/test.in
 django-crum==0.7.5        # via edx-rbac
-django-fernet-fields==0.5
-django-model-utils==3.2.0
-django-waffle==0.18.0     # via edx-django-utils, edx-drf-extensions
-django==1.11.15
+django-fernet-fields==0.5  # via -r requirements/base.in
+django-model-utils==3.2.0  # via -r requirements/base.in, -r requirements/test.in, edx-rbac
+django-waffle==0.20.0     # via edx-django-utils, edx-drf-extensions
+django==1.11.15           # via -r requirements/base.in, -r requirements/test-master.in, django-crum, django-fernet-fields, django-model-utils, edx-django-utils, edx-drf-extensions, edx-rbac, rest-condition
```
Comment (jmbowman) [inline requirements/test-master.txt]: It looks like Django is pinned to this version in `test-master.in` - it shouldn't be.  It should be allowed to use the latest release < 2.0.

File: requirements/test-master.txt [hunk 5]
```diff
@@ -8,101 +8,98 @@ amqp==1.4.9               # via kombu
...
+pymongo==3.10.1           # via edx-opaque-keys
 pynacl==1.3.0             # via paramiko
-pyopenssl==17.4.0
+pyopenssl==17.4.0         # via -r requirements/base.in
 pyparsing==2.4.6          # via packaging
-pytest-catchlog==1.2.2
-pytest-cov==2.8.1
-pytest-django==3.7.0
-pytest==4.6.8             # via pytest-catchlog, pytest-cov, pytest-django
+pytest-catchlog==1.2.2    # via -r requirements/test.in
```
Comment (jmbowman) [inline requirements/test-master.txt]: This isn't needed anymore, it was merged into pytest 3.3.0.

**Review by jmbowman — COMMENTED** [review]
I spotted a few minor problems I missed on the first pass, should be good after that.  Could you also add an entry under "Unreleased" in the changelog?  Then the next PR can update it with further changes and bump the version number.

**Comment by morenol** [comment]
I made those changes @jmbowman. I have in mind other changes to the requirements, for instance https://github.com/edx/edx-enterprise-data/pull/166. Should I add these changes in a different PR or should I add them in this one?

**Review by jmbowman — APPROVED** [review]
I'll go ahead and merge this one, please rebase your other dependency PRs to get these changes.  It'll be easier to review the other changes once these minor upgrades and pip-tools output format update are in place.

**Comment by openedx-webhooks** [comment]
@morenol 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/edx-enterprise-data&entry.867055334=https://github.com/edx/edx-enterprise-data/pull/165&entry.1484655318=https://github.com/morenol&entry.752974735=2020-03-02+14:57&entry.1917517419=2020-03-03+17:55&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**